bundles/org.eclipse.swt/buildSWT.xml

```diff
@@ -291,19 +291,27 @@
 		</exec>
 		<script language="javascript">
 			<![CDATA[
-				tags = project.getProperty("tags");
-				swt_version = project.getProperty("swt_version");
-				if (!tags.match(swt_version)) {
-					project.setProperty("swt_new_tag", swt_version);
-				} else {
-					for (i=97; i<123; i++) {
-						t = swt_version + String.fromCharCode(i);
-						if (!tags.match(t)) {
-							project.setProperty("swt_new_tag", t);
-							break;
-						}
-					}
-				}
+	        tags = project.getProperty("tags");
+	        swt_version = project.getProperty("swt_version");
+	        if (!tags.match(swt_version)) {
+	            project.setProperty("swt_new_tag", swt_version);
+	        } else {
+	            var found = false;
+	            var suffix = 'a';
+	            while (!found) {
+	                var t = swt_version + suffix;
+	                if (!tags.match(t)) {
+	                    project.setProperty("swt_new_tag", t);
+	                    found = true;
+	                } else {
+	                    if (suffix === 'z') {
+	                        suffix = 'a'; // Reset to 'a' after 'z'
```

Comment:
In existing code, if we've reached the case of 'z', it is most likely that the case of 'a' (`i=97`) and all other letters already reported that the suffix is already used.
You would basically want to write an algorithm that iterates through the sequences a..z,aa...az, ba...bz, ..., za...zz, aaa...aaz, aba...abz, ..., aza,...,azz, baa...baz and so on. It's a lot of smartness to put in a build scripts, particularly if we don't really need the tags.
Maybe just a timestamp suffix would be enough?

Reply:
Does github support numbers in suffixes ?

Reply:
Git and GitHub don't have a concept of suffix, it's just some characters in the tag name; and yes, they do support numbers as characters in tag names.

Reply:
We would'nt have this problem if the v4963r3 part of the tags were sure to be updated before 26 commits,
What mechanism updates this part or the tag ?